src/mon/OSDMonitor.cc

```diff
@@ -6759,8 +6759,7 @@ int OSDMonitor::prepare_command_pool_set(const cmdmap_t& cmdmap,
       return r;
     }
     p.size = n;
-    if (n < p.min_size)
-      p.min_size = n;
+    p.min_size = g_conf->get_osd_pool_default_min_size();
```

Comment:
not sure, but maybe in conflict resolution code might had got updated, it is missing the pool size being passed. The master pull request has:  
```suggestion
    p.min_size = g_conf->get_osd_pool_default_min_size(p.size);
```